pkg/autoscaler/aggregation/bucketing.go

```diff
@@ -24,43 +24,51 @@ import (
 	"github.com/davecgh/go-spew/spew"
 )
 
-// TimedFloat64Buckets keeps buckets that have been collected at a certain time.
-type TimedFloat64Buckets struct {
-	bucketsMutex sync.RWMutex
-	// buckets is a ring buffer indexed by timeToIndex() % len(buckets).
-	// Each element represents a certain granularity of time, and the total
-	// represented duration adds up to a window length of time.
-	buckets []float64
-
-	// firstWrite holds the time when the first write has been made.
-	// This time is reset to `now` when the very first write happens,
-	// or when a first write happens after `window` time of inactivity.
-	// The difference between `now` and `firstWrite` is used to compute
-	// the number of eligible buckets for computation of average values.
-	firstWrite time.Time
-
-	// lastWrite stores the time when the last write was made.
-	// This is used to detect when we have gaps in the data (i.e. more than a
-	// granularity has expired since the last write) so that we can zero those
-	// entries in the buckets array. It is also used when calculating the
-	// WindowAverage to know how much of the buckets array represents valid data.
-	lastWrite time.Time
-
-	// granularity is the duration represented by each bucket in the buckets ring buffer.
-	granularity time.Duration
-	// window is the total time represented by the buckets ring buffer.
-	window time.Duration
-	// The total sum of all buckets within the window. This total includes
-	// invalid buckets, e.g. buckets written to before firstTime or after
-	// lastTime are included in this total.
-	windowTotal float64
-
-	// decayMultiplier contains the speed with which the importance
-	// of items in the past decays. The larger the faster weights decay.
-	// It is autocomputed from window size and weightPrecision constant
-	// and is bounded by minExponent below.
-	decayMultiplier float64
-}
+type (
```

Comment:
I dont know how I feel about this, we don't do it anywhere else with `type` afaik, should we?

Reply:
we do, in a few places :)